zetaclient/chains/sui/signer/signer_tx.go

```diff
@@ -147,28 +147,56 @@ func (s *Signer) buildWithdrawAndCallTx(
 		return models.TxnMetaData{}, errors.Wrap(err, "unable to parse withdrawAndCall payload")
 	}
 
-	// Note: logs not formatted in standard, it's a temporary log
-	s.Logger().Std.Info().Msgf(
-		"WithdrawAndCall called with type arguments %v, object IDs %v, message %v",
-		cp.TypeArgs,
+	// get latest TSS SUI coin object ref for gas payment
+	suiCoinObjRef, err := s.client.GetSuiCoinObjectRef(ctx, s.TSS().PubKey().AddressSui())
+	if err != nil {
+		return models.TxnMetaData{}, errors.Wrap(err, "unable to get TSS SUI coin object")
+	}
+
+	// get all other object references: [gateway, withdrawCap, onCallObjects]
+	gatewayObjRef, withdrawCapObjRef, onCallObjectRefs, err := getWithdrawAndCallObjectRefs(
+		ctx,
+		s.client,
+		s.gateway.ObjectID(),
+		withdrawCapIDStr,
 		cp.ObjectIDs,
-		cp.Message,
 	)
+	if err != nil {
+		return models.TxnMetaData{}, errors.Wrap(err, "unable to get object references")
+	}
 
-	// keep lint quiet without using _ in params
-	_ = ctx
-	_ = params
-	_ = coinType
-	_ = gasBudget
-	_ = withdrawCapID
+	// print PTB transaction parameters
+	s.Logger().Std.Info().
+		Str(logs.FieldMethod, "buildWithdrawAndCallTx").
+		Str(logs.FieldCoinType, coinType).
+		Str("amount", params.Amount.String()).
+		Uint64(logs.FieldNonce, params.TssNonce).
+		Str("receiver", params.Receiver).
+		Str("gas_budget", gasBudget).
+		Any("type_args", cp.TypeArgs).
+		Any("object_ids", cp.ObjectIDs).
+		Hex("message", cp.Message).
+		Msg("calling withdrawAndCallPTB")
 
 	// TODO: check all object IDs are share object here
 	// https://github.com/zeta-chain/node/issues/3755
```

Comment:
We now have the function that we can put here
```
func (c *Client) CheckObjectIDsShared(ctx context.Context, objectIDs []string) error 
```

Reply:
https://github.com/zeta-chain/node/pull/3793/commits/2376ca94a094605cabd68cbb17a2be4849762532